### PR TITLE
chore(pro): thread through errors for oom and timeout during preprocessing

### PR DESCRIPTION
**Pull request by emjin**
This will help us keep track of which errors are raised by pro engine
users that prevent pro analysis from happening. Also, sets a default
pro timeout for CI users

See https://github.com/returntocorp/semgrep-proprietary/pull/465 for more information

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/src/semgrep/commands/scan.py
Comment (nmote) [inline cli/src/semgrep/commands/scan.py]: That's a bit confusing

**Review by nmote — APPROVED** [review]
Looks good! This will be a breaking change though, right? New versions of the CLI will no longer function with even slightly older versions of semgrep-proprietary.

**Comment by nmote** [comment]
Please don't merge until you're ready to merge the associated changes to semgrep-proprietary

File: cli/src/semgrep/commands/scan.py
Comment (IagoAbal) [inline cli/src/semgrep/commands/scan.py]: Maybe App should rename the toggle to `Semgrep Pro` (even if it's just in the UI, that should be easy) so that we don't need to refer to "DeepSemgrep" here either. I also find  a bit confusing to refer to the App toggle here. This is the `scan` command command, not the `ci` one.

File: cli/src/semgrep/commands/scan.py
Comment (IagoAbal) [inline cli/src/semgrep/commands/scan.py]: As I said in the other PR I think this should be "inter-file analysis". Apex is a Pro feature and doesn't need this timeout. And `--interproc` won't need this timeout either I think.

**Review by IagoAbal — APPROVED** [review]

File: cli/src/semgrep/commands/scan.py
Comment (emjin) [reply cli/src/semgrep/commands/scan.py]: Yeah, I agree that's a bit confusing but I don't know another place to put it

File: cli/src/semgrep/commands/scan.py
Comment (emjin) [reply cli/src/semgrep/commands/scan.py]: Made some clarifications

**Pull request merged**
